lightning/src/ln/channelmanager.rs

```diff
@@ -4990,6 +5008,7 @@ where
 		// This ensures that future code doesn't introduce a lock-order requirement for
 		// `forward_htlcs` to be locked after the `per_peer_state` peer locks, which calling
 		// this function with any `per_peer_state` peer lock acquired would.
+		#[cfg(debug_assertions)]
```

Comment:
`fn internal_shutdown` seems to call `finish_force_close_channel` with the peer_state lock held.